src/api/apiMethods.js

```diff
@@ -0,0 +1,324 @@
+        fixedColumns,
+        fixedColumnGroups,
+        fixedRightColumns,
+        fixedRightColumnGroups,
+        colOffsetIntervalTree,
+        fixedColumnOffsets,
+        fixedColumnGroupOffsets,
+        fixedRightColumnOffsets,
+        fixedRightColumnGroupOffsets,
+        storedScrollableColumns,
+        storedScrollableColumnGroups,
+      } = state;
+
+      const getColumnOffset = (index, cellGroupType) => {
+        if (cellGroupType === 'fixed') {
+          fixedColumnOffsets[index];
+        } else if (cellGroupType === 'fixedRight') {
+          fixedRightColumnOffsets[index];
+        } else {
+          return colOffsetIntervalTree.sumTo(index);
+        }
+      };
+
+      const getCellGroupWidth = (cellGroupType = 'scrollable') => {
+        if (cellGroupType === 'fixed') {
+          return fixedColumnsWidth;
+        } else if (cellGroupType === 'fixedRight') {
+          return fixedRightColumnsWidth;
+        } else {
+          return scrollContentWidth;
+        }
+      };
+
+      const _getColumn = (columnIndex, cellGroupType = 'scrollable') => {
+        if (cellGroupType === 'fixed') {
+          return fixedColumns[columnIndex];
+        } else if (cellGroupType === 'fixedRight') {
+          return fixedRightColumns[columnIndex];
+        } else {
+          return _getScrollableColumn(columnIndex);
+        }
+      };
+
+      const _getScrollableColumn = (columnIndex) => {
+        let column = storedScrollableColumns.object[columnIndex];
+
+        // if no column exists in the cache, then fetch the column from the user
+        if (!column) {
+          column = convertColumnElementsToData(
+            columnSettings.getScrollableColumn(columnIndex)
+          );
+
+          // we don't store the column to the cache because our functions should be pure
+          // this also means that we need to keep FDT's default width for the column
+          column.props.width = columnSettings.defaultColumnWidth;
+        }
+
+        return column;
+      };
+
+      const _getScrollableColumnGroup = (columnGroupIndex) => {
+        // TODO (pradeep): Clone deep might cause performance issues
+        let columnGroup = storedScrollableColumnGroups.object[columnGroupIndex];
+
+        // if no column group exists in the cache, then fetch the column group from the user
+        if (!columnGroup) {
+          columnGroup = convertColumnElementsToData(
+            columnSettings.getScrollableColumnGroup(columnGroupIndex)
+          );
+
+          // we also need to figure out the children of the column group to accurately determine the start and end child indexes, as well as the supposed total width of the column group
+          let firstChildIndex;
+          let lastChildIndex;
+          // NOTE (pradeep): This runs at O(N). Consider another approach, like a binary search to atleast make this run at O(log N).
+          for (let i = 0; i < columnSettings.scrollableColumnsCount; i++) {
```

Comment:
If the column group is not present in the cache, then figuring out the first and child column indices is not straightforward.
We'll have to first figure out what column has it's `columnGroupIndex` set to the given column group. 
For now I've put here a very simple iterative loop which runs at O(N). But we can make this more efficient later on by using a binary search.